### PR TITLE
Implement user state saving for projects overview [SCI-2733]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-2733

Datatables were also updated to recent version, some unused plugins removed

### CODE REVIEW

File: app/models/view_state.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal:true
+
+class ViewState < ApplicationRecord
+  belongs_to :user
```
Comment (Ducz0r) [inline app/models/view_state.rb]: Nice, simple & efficient :+1:.

File: app/models/team.rb [hunk 2]
```diff
@@ -35,11 +35,28 @@ class Team < ApplicationRecord
   has_many :reports, inverse_of: :team, dependent: :destroy
   has_many :datatables_reports,
            class_name: 'Views::Datatables::DatatablesReport'
+  has_many :view_states, as: :viewable, dependent: :destroy
 
   after_commit do
     Views::Datatables::DatatablesReport.refresh_materialized_view
   end
 
+  def current_view_state(user)
```
Comment (Ducz0r) [inline app/models/team.rb]: I'm just thinking, could we isolate this (albeit not very big) code block into its own service, maybe even an abstracted one (e.g. have an abstract service interface for retrieving - and possibly saving - state for any user - object combo)?

I'm generally not a fan of Ruby's fat models paradigm (after seeing how bloated things can become). I just feel that this various state-keeping logic for GUI components doesn't really belong scattered into multiple `ActiveRecord` data classes. If anything, I myself conceptually relate these states more to users than to their related objects. That's why I think isolated services might be the most clean solution.

What do you think?

File: app/models/team.rb [hunk 3]
```diff
@@ -35,11 +35,28 @@ class Team < ApplicationRecord
...
+    state = view_states.where(user: user).take
+    state || view_states.create!(user: user, state: default_view_state)
+  end
+
+  def default_view_state
```
Comment (Ducz0r) [inline app/models/team.rb]: I just have a question: let's say that we also implement state save for e.g. individual team settings page (where there's a `DataTables`  listing all the users in a team). In this case, we're also storing a state for a `current_user` - `current_team` combo. Wouldn't it make sense to add additional root-level to this JSONB to designate which "view" for the user-object relation the state belongs to?

File: app/services/projects_overview_service.rb [hunk 4]
```diff
@@ -118,15 +135,19 @@ def sortable_columns
...
+    order = @params[:order]&.values&.first
     if order
       dir = order[:dir] == 'desc' ? 'DESC' : 'ASC'
       column_index = order[:column]
     else
       dir = 'ASC'
       column_index = '1'
     end
+    if order_state != [column_index.to_i, dir.downcase]
+      @view_state.state['table']['order'][0] = [column_index.to_i, dir.downcase]
```
Comment (Ducz0r) [inline app/services/projects_overview_service.rb]: Is this line totally correct? In default state, the order is like this: `order: [2, 'asc']`, but here, it changes to `order: [[2, 'asc'], 'asc']`.

Comment (okriuchykhin) [reply app/services/projects_overview_service.rb]: Yes, this is correct, order should contain two-dimensional array, I will fix default value

File: app/models/team.rb (hunk 3 above)
Comment (okriuchykhin) [reply app/models/team.rb]: Yeah, it makes sense

File: app/models/team.rb (hunk 2 above)
Comment (okriuchykhin) [reply app/models/team.rb]: I can move it ot Concern

**Pull request merged**